### PR TITLE
fix: error message for trust policy

### PR DESCRIPTION
**Pull request by JeyJeyGao**
This error message improvement is based on the Notation CLI error message improvement guideline that we defined in #834 

Resolve part of #934 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: cmd/notation/policy/import.go [hunk 1]
```diff
@@ -42,7 +42,12 @@ func importCmd() *cobra.Command {
 Example - Import trust policy configuration from a file:
   notation policy import my_policy.json
 `,
-		Args: cobra.ExactArgs(1),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("requires 1 argument but received %d.\nUsage: notation policy import <path-to-policy.json>\nPlease specify a trust policy file as the argument", len(args))
+			}
+			return nil
```
Comment (Two-Hearts) [inline cmd/notation/policy/import.go:49]: Do we have E2E test regarding this change? If not, please add it.

File: cmd/notation/policy/show.go [hunk 2]
```diff
@@ -59,7 +59,10 @@ func runShow(command *cobra.Command, opts showOpts) error {
 	// core process
 	policyJSON, err := os.ReadFile(policyPath)
 	if err != nil {
-		return fmt.Errorf("failed to load trust policy configuration, you may import one via `notation policy import <path-to-policy.json>`: %w", err)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("failed to show trust policy configuration because the trust policy file does not exist.\nYou may import one via `notation policy import <path-to-policy.json>`")
+		}
+		return fmt.Errorf("failed to load trust policy configuration: %w", err)
```
Comment (Two-Hearts) [inline cmd/notation/policy/show.go:65]: line 65 is having a different pattern than line 63. Maybe line 65 should also be:
```suggestion
		return fmt.Errorf("failed to show trust policy: %w", err)
```
And looks like line 65 is not covered by any test, please add it.

File: cmd/notation/policy/import.go (hunk 1 above)
Comment (JeyJeyGao) [reply cmd/notation/policy/import.go:49]: Updated.

File: cmd/notation/policy/show.go (hunk 2 above)
Comment (JeyJeyGao) [reply cmd/notation/policy/show.go]: Updated.

File: cmd/notation/policy/show.go [hunk 3]
```diff
@@ -53,13 +53,16 @@ func runShow(command *cobra.Command, opts showOpts) error {
...
 	if err != nil {
-		return fmt.Errorf("failed to obtain path of trust policy configuration file: %w", err)
+		return fmt.Errorf("failed to obtain path of trust policy file: %w", err)
 	}
 
 	// core process
 	policyJSON, err := os.ReadFile(policyPath)
 	if err != nil {
-		return fmt.Errorf("failed to load trust policy configuration, you may import one via `notation policy import <path-to-policy.json>`: %w", err)
+		if os.IsNotExist(err) {
```
Comment (shizhMSFT) [inline cmd/notation/policy/show.go:62]: nit: New code should use `errors.Is(err, fs.ErrNotExist)`.

See https://pkg.go.dev/os#IsNotExist

**Review by shizhMSFT — DISMISSED** [review]
LGTM with suggestions

File: cmd/notation/policy/show.go (hunk 3 above)
Comment (JeyJeyGao) [reply cmd/notation/policy/show.go]: Thanks! Updated.

File: cmd/notation/policy/show.go [hunk 4]
```diff
@@ -53,13 +55,16 @@ func runShow(command *cobra.Command, opts showOpts) error {
 	// get policy file path
 	policyPath, err := dir.ConfigFS().SysPath(dir.PathTrustPolicy)
 	if err != nil {
-		return fmt.Errorf("failed to obtain path of trust policy configuration file: %w", err)
+		return fmt.Errorf("failed to obtain path of trust policy file: %w", err)
```
Comment (JeyJeyGao) [inline cmd/notation/policy/show.go:58]: SysPath() doesn't return an error

File: test/e2e/suite/command/policy.go [hunk 5]
```diff
@@ -60,7 +70,9 @@ var _ = Describe("trust policy maintainer", func() {
 		It("should fail if no file path is provided", func() {
```
Comment (Two-Hearts) [inline test/e2e/suite/command/policy.go:70]: Can we add another test case with more than one argument? For example, two trust policy files are provided.

Comment (JeyJeyGao) [reply test/e2e/suite/command/policy.go:70]: Updated.

**Review by Two-Hearts — DISMISSED** [review]
LGTM

File: cmd/notation/policy/show.go [hunk 6]
```diff
@@ -53,13 +55,16 @@ func runShow(command *cobra.Command, opts showOpts) error {
...
 	}
 
 	// core process
 	policyJSON, err := os.ReadFile(policyPath)
 	if err != nil {
-		return fmt.Errorf("failed to load trust policy configuration, you may import one via `notation policy import <path-to-policy.json>`: %w", err)
+		if errors.Is(err, fs.ErrNotExist) {
+			return fmt.Errorf("failed to show trust policy because the trust policy file does not exist.\nYou may import one via `notation policy import <path-to-policy.json>`")
```
Comment (priteshbandi) [inline cmd/notation/policy/show.go:65]:
```suggestion
			return fmt.Errorf("failed to show trust policy as the trust policy file does not exist.\nYou can import one using `notation policy import <path-to-policy.json>`")
```

File: cmd/notation/policy/import.go [hunk 7]
```diff
@@ -42,7 +42,12 @@ func importCmd() *cobra.Command {
 Example - Import trust policy configuration from a file:
   notation policy import my_policy.json
 `,
-		Args: cobra.ExactArgs(1),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return fmt.Errorf("requires 1 argument but received %d.\nUsage: notation policy import <path-to-policy.json>\nPlease specify a trust policy file as the argument", len(args))
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply cmd/notation/policy/import.go]: Updated.

File: cmd/notation/policy/import.go [hunk 8]
```diff
@@ -71,7 +76,7 @@ func runImport(command *cobra.Command, opts importOpts) error {
 	// optional confirmation
 	if !opts.force {
 		if _, err := trustpolicy.LoadDocument(); err == nil {
-			confirmed, err := cmdutil.AskForConfirmation(os.Stdin, "Existing trust policy configuration found, do you want to overwrite it?", opts.force)
+			confirmed, err := cmdutil.AskForConfirmation(os.Stdin, "The trust policy configuration already exists, do you want to overwrite it?", opts.force)
```
[the remark this answers is not in the document]
Comment (JeyJeyGao) [reply cmd/notation/policy/import.go]: Updated.

File: cmd/notation/policy/show.go (hunk 6 above)
Comment (JeyJeyGao) [reply cmd/notation/policy/show.go]: Updated.

**Review by Two-Hearts — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**